registry/storage/graph.go

```diff
@@ -0,0 +1,600 @@
+package storage
+
+import (
+	"container/list"
+	"fmt"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/context"
+	"github.com/docker/distribution/digest"
+	"github.com/docker/distribution/manifest/schema1"
+)
+
+// graph contains functions for collecting information about orphaned objects
+// and dangling links.
+// These functions will only reliably work on strongly consistent storage
+// systems.
+// https://en.wikipedia.org/wiki/Consistency_model
+
+// Empty is a type of set values using no space.
+type Empty struct{}
```

Comment:
Needless for export or definition. Just use `struct{}`. This is a common and well known Go pattern.